features/step_definitions/data.js

```diff
@@ -48,41 +48,26 @@ module.exports = function () {
         q.awaitAll(callback);
     });
 
-    this.Given(/^the node map$/, (table, callback) => {
-        let q = d3.queue();
-
-        let addNode = (name, ri, ci, cb) => {
-            if (name) {
-                let nodeWithID = name.match(/([a-z])\:([0-9]*)/);
-                if (nodeWithID) {
-                    let nodeName = nodeWithID[1],
-                        nodeID = nodeWithID[2];
-                    if (this.nameNodeHash[nodeName]) throw new Error(util.format('*** duplicate node %s', name));
-                    let lonLat = this.tableCoordToLonLat(ci, ri);
-                    this.addOSMNode(nodeName, lonLat[0], lonLat[1], nodeID);
-                } else {
-                    if (name.length !== 1) throw new Error(util.format('*** node invalid name %s, must be single characters', name));
-                    if (!name.match(/[a-z0-9]/)) throw new Error(util.format('*** invalid node name %s, must me alphanumeric', name));
-
-                    if (name.match(/[a-z]/)) {
-                        if (this.nameNodeHash[name]) throw new Error(util.format('*** duplicate node %s', name));
-                        let lonLat = this.tableCoordToLonLat(ci, ri);
-                        this.addOSMNode(name, lonLat[0], lonLat[1], null);
-                    } else {
-                        if (this.locationHash[name]) throw new Error(util.format('*** duplicate node %s'), name);
-                        let lonLat = this.tableCoordToLonLat(ci, ri);
-                        this.addLocation(name, lonLat[0], lonLat[1], null);
-                    }
-                }
-
-                cb();
+    this.Given(/^the node map$/, (docstring, callback) => {
+        var q = d3.queue();
+
+        var addNode = (name, ri, ci, cb) => {
+            var lonLat = this.tableCoordToLonLat(ci, ri);
+            if (name.match(/[a-z]/) ) {
+                if (this.nameNodeHash[name]) throw new Error(util.format('*** duplicate node %s', name));
+                this.addOSMNode(name, lonLat[0], lonLat[1], null);
+            } else if (name.match(/[0-9]/) ) {
+                if (this.locationHash[name]) throw new Error(util.format('*** duplicate node %s'), name);
+                this.addLocation(name, lonLat[0], lonLat[1], null);
             }
-            else cb();
+            cb();
         };
 
-        table.raw().forEach((row, ri) => {
-            row.forEach((name, ci) => {
-                q.defer(addNode, name, ri, ci);
+        docstring.split(/\n/).forEach( (row,ri) => {
+            row.split('').forEach( (cell,ci) => {
+                if( cell.match(/[a-z0-9]/) ) {
+                    q.defer(addNode, cell, ri, ci*0.5);
```

Comment:
Yes. This is to ensure that the map doesn't look distorted (this was part of the original intention behind the PR). eg. that a square will look like a square and not a rectangle. Compare:

```gherkin
And the node map
    """
    a . b
    .   .
    c . d
    """

And the node map
    """
    a.b
    . .
    c.d
    """
```

With the current implementation, you're allowed to place nodes anywhere, so horizontally you can place nodes with 0.5 times the grid distance, vertically only on the grid. The following is valid and will result in distance of ac=1, cd=1, but ab=0.5:

```gherkin
And the node map
    """
    ab
    c d
    """
```

An alternative would be to enforce placement only on the grid, ie. always have spacing between nodes horizontally.